src/devices/cddadevice.cpp

```diff
@@ -36,24 +40,53 @@ CddaDevice::CddaDevice(const QUrl& url, DeviceLister* lister,
           SLOT(SongsLoaded(SongList)));
   connect(this, SIGNAL(SongsDiscovered(SongList)), model_,
           SLOT(SongsDiscovered(SongList)));
+  //   connect(&cd_device_, SIGNAL(DiscChanged()), SLOT(DiscChangeDetected()));
+  cdio_ = cdio_open(url_.path().toLocal8Bit().constData(), DRIVER_DEVICE);
+  if (!cdio_) {
+    throw DeviceError(url.toString(),
+                      "Cannot open device: cdio_open returned nullptr");
```

Comment:
Exceptions aren't used for error handling anywhere else in the code base. It'll be up to @hatstand to decide if he wants to introduce them.

Reply:
I'd argue it's the best way to do this here: There is no other way to pass out an error condition, neither in the constructor nor in the `Init` currently. So we could have an error code returned by `Init`, which would mean touching all classes implementing `ConnectedDevice`. Alternatively we could (re)introduce an `IsValid` method for `CddaDevice`. Checking that after instantiation would then make it essentially an exception in disguise (and also be more error prone, as it would be possible to forget checking that). We could also check validity every time the object would be about to be used, but at that point we're just looking for ways to make stuff complicated.

Reply:
Sorry, please avoid exceptions.

Reply:
alright.. done. Have modified `Init()` to return a boolean indicating success. Please see if this is fine.